src/hpc/serialize.c

```diff
@@ -775,41 +736,41 @@ static Obj LookupIntTag(Obj tag)
     }
 }
 
-static Obj DeserializeTypedObj(UInt tnum)
+static Obj DeserializeTypedObj(DeserializerState * state, UInt tnum)
 {
     UInt namelen, len, i;
     Obj  name, args, deserialization_rec, func, type, tag;
     Obj  result;
     UInt rnam, tagtnum;
-    tagtnum = ReadTNum();
+    tagtnum = ReadTNum(state);
     switch (tagtnum) {
     case T_INT:
     case T_STRING:
     case T_STRING + IMMUTABLE:
         if (tagtnum == T_INT) {
-            tag = DeserializeInt(T_INT);
+            tag = DeserializeInt(state, T_INT);
             type = LookupIntTag(tag);
         }
         else {
-            tag = DeserializeString(T_STRING);
+            tag = DeserializeString(state, T_STRING);
             rnam = RNamObj(tag);
             type = ELM_REC(GVarObj(&DESERIALIZATION_TAG_STRING_GVar), rnam);
         }
         if (!type || TNUM_OBJ(type) != T_POSOBJ)
             DeserializationError();
         switch (tnum) {
         case T_DATOBJ:
-            len = ReadByteBlockLength();
+            len = ReadByteBlockLength(state);
             result = NewBag(T_DATOBJ, len + sizeof(Obj));
-            ReadByteBlockData(result, sizeof(Obj), len);
+            ReadByteBlockData(state, result, sizeof(Obj), len);
             break;
         case T_POSOBJ:
-            result = DeserializeObj();
+            result = DeserializeObj(state);
             if (!IS_PLIST(result))
                 DeserializationError();
             break;
         case T_PREC:
-            result = DeserializeObj();
+            result = DeserializeObj(state);
```

Comment:
Codecov says this line and some others in this function aren't covered by tests. Would it be relatively easy to do so and, if you think it's worth the time, would you add some?

Reply:
No, I won't: all the existing tests were added by me (i.e. there were zero before I did that). Whenever I add tests to this code, I uncover bugs, and then I won't be able to stop myself from trying to fix them.

As it is, I only touched this code right now because I needed to understand whether the GAP_TRY/GAP_CATCH did anything "interesting". I was able to eliminate them, which is good, as that's one place less I have to worry about...

(On the long run, we should either discard this code along with the rest of HPC-GAP, or come up with a plan to start using them. The serializer code might be of interest independent of HPC-GAP, as an alternative to IO pickling that works on non-POSIX systems, and is built into GAP itself...)